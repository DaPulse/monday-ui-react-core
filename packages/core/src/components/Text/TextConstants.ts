/**
 * @deprecated
 */
export enum TextType {
  TEXT1 = "text1",
  TEXT2 = "text2",
  TEXT3 = "text3"
}

/**
 * @deprecated
 */
export enum TextWeight {
  BOLD = "bold",
  MEDIUM = "medium",
  NORMAL = "normal"
}
