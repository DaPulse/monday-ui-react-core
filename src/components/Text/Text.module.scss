@import "../../styles/typography";

@include create-text-classes;
