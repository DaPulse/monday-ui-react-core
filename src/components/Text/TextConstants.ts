export type TextSize = "medium" | "small";

export type TextWeight = "bold" | "normal";
