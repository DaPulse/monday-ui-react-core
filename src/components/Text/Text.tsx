import React, { forwardRef } from "react";
import cx from "classnames";
import VibeComponent from "../../types/VibeComponent";
import { TextSize, TextWeight } from "./TextConstants";
import Typography, { TypographyProps } from "../Typography/Typography";
import styles from "./Text.module.scss";

export interface TextProps extends TypographyProps {
  size?: TextSize;
  weight?: TextWeight;
  paragraph?: boolean;
}

const Text: VibeComponent<TextProps, HTMLElement> = forwardRef(
  ({ className, size = "medium", weight = "normal", ellipsis, paragraph, ...typographyProps }, ref) => {
    const overrideEllipsis = ellipsis ?? !paragraph;
    const overrideElement = paragraph ? "p" : "div";
    return (
      <Typography
        ref={ref}
        className={cx(styles[size], styles[weight], className)}
        ellipsis={overrideEllipsis}
        element={overrideElement}
        {...typographyProps}
      />
    );
  }
);

export default Text;
