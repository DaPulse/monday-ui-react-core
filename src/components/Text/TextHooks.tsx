import { TextSize, TextWeight } from "./TextConstants";
import { MutableRefObject, useEffect } from "react";
import styles from "./Text.module.scss";

export function useGlobalTextClass(size: TextSize, weight: TextWeight) {
  return `vibe-text${size == TextSize.MEDIUM ? 1 : 2}-${weight}`;
}

export function useEllipsisClass(ref: MutableRefObject<HTMLElement>, ellipsis: boolean, maxLines: number) {
  // Set css variable on component mount for supporting multiple line ellipsis
  useEffect(() => {
    ref.current.style.setProperty("--text-clamp-lines", maxLines.toString());
  });

  // If component contains ellipsis return the fit ellipsis class
  if (!ellipsis) return;
  return maxLines > 1 ? styles.multiLineEllipsis : styles.singleLineEllipsis;
}
