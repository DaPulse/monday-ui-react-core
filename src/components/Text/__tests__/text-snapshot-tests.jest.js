import React from "react";
import renderer from "react-test-renderer";
import Text from "../Text";

describe("Text renders correctly", () => {
  it("with empty props", () => {
    const tree = renderer.create(<Text />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with size", () => {
    const tree = renderer.create(<Text size={Text.sizes.SMALL}>text</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with weight", () => {
    const tree = renderer.create(<Text weight={Text.weights.BOLD}>text</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with color", () => {
    const tree = renderer.create(<Text color={Text.colors.SECONDARY}>text</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with element", () => {
    const tree = renderer.create(<Text element="p">text</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with align", () => {
    const tree = renderer.create(<Text align={Text.align.CENTER}>text</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with element as children", () => {
    const tree = renderer
      .create(
        <Text>
          <div>text</div>
        </Text>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
