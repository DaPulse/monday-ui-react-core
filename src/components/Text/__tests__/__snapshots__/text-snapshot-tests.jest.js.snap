// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly with align 1`] = `
<div
  className="typography primary center singleLineEllipsis mediumNormal"
  data-testid="text"
>
  text
</div>
`;

exports[`Text renders correctly with color 1`] = `
<div
  className="typography secondary start singleLineEllipsis mediumNormal"
  data-testid="text"
>
  text
</div>
`;

exports[`Text renders correctly with element 1`] = `
<p
  className="typography primary start singleLineEllipsis mediumNormal"
  data-testid="text"
>
  text
</p>
`;

exports[`Text renders correctly with element as children 1`] = `
<div
  className="typography primary start singleLineEllipsis mediumNormal"
  data-testid="text"
>
  <div>
    text
  </div>
</div>
`;

exports[`Text renders correctly with empty props 1`] = `
<div
  className="typography primary start singleLineEllipsis mediumNormal"
  data-testid="text"
/>
`;

exports[`Text renders correctly with paragraph 1`] = `
<p
  className="typography primary start mediumNormal"
  data-testid="text"
>
  text
</p>
`;

exports[`Text renders correctly with size 1`] = `
<div
  className="typography primary start singleLineEllipsis smallNormal"
  data-testid="text"
>
  text
</div>
`;

exports[`Text renders correctly with weight 1`] = `
<div
  className="typography primary start singleLineEllipsis mediumBold"
  data-testid="text"
>
  text
</div>
`;
